tmt/utils/__init__.py

```diff
@@ -4325,20 +4330,19 @@ def url_and_name(self, cwd: Optional[Path] = None) -> list[tuple[str, str]]:
         package = globbed[0].stem
         ret_values = []
         try:
-            with open(cwd / self.sources_file_name) as f:
-                for line in f:
-                    match = self.re_source.match(line)
-                    if match is None:
-                        raise GeneralError(
-                            f"Couldn't match '{self.sources_file_name}' "
-                            f"content with '{self.re_source.pattern}'.")
-                    used_hash, source_name, hash_value = match.groups()
-                    ret_values.append((self.lookaside_server + self.uri.format(
-                        name=package,
-                        filename=source_name,
-                        hash=hash_value,
-                        hashtype=used_hash.lower()
-                        ), source_name))
+            for line in (cwd / self.sources_file_name).read_text().splitlines():
```

Comment:
Seems there is a pattern here, could we add

```
def read_lines(path: Path) -> list[str]:
    return path.read_text().splitlines()
```

To our utils? So it would become:

```
            for line in read_path_lines(cwd / self.sources_file_name):
```

Reply:
Added in 187f79de. Iterator is not possible because both `read_text()` and `splitlines()` have their own options.

Reply:
ack, looks great to me.